### PR TITLE
Feature/gf 1234 previewactslikesave

### PR DESCRIPTION
**Pull request by BenKDishman**
This is a small change to allow marking as complete on preview.  Probably only needs 1 Approval and has no corresponding business layer  change.  Only real question I have is whether ActionHandler::PreviewAndComplete is a little convoluted.  Seems okay to me as it is pulling in various existing functions into 1 place but I thought I'd ask the question!

### CODE REVIEW

File: src/main/scala/controllers/ApplicationController.scala [hunk 1]
```diff
@@ -83,6 +84,7 @@ class ApplicationController @Inject()(actionHandler: ActionHandler, applications
       case Save => actionHandler.doSave(id, sectionNumber, request.body.values)
       case SaveItem => actionHandler.doSaveItem(id, sectionNumber, request.body.values)
       case Preview => actionHandler.doPreview(id, sectionNumber, request.body.values)
+      case PreviewAndComplete => actionHandler.PreviewAndComplete(id, sectionNumber, request.body.values)
```
Comment (DougC) [inline src/main/scala/controllers/ApplicationController.scala]: Would suggest naming it CompleteAndPreview, as that's the order in which things happen.

File: src/main/scala/controllers/ButtonAction.scala [hunk 2]
```diff
@@ -20,13 +22,18 @@ case object SaveItem extends ButtonAction {
   override val name: String = "_save_item_button"
 }
 
+case object CompleteAndPreview extends ButtonAction {
+  override val name: String = "_preview_and_complete_button"
```
Comment (DougC) [inline src/main/scala/controllers/ButtonAction.scala]: I know it's not actually used, but maybe change this to match the name of the case object (i.e. "_complete_and_preview_button"

File: src/main/scala/controllers/ActionHandler.scala [hunk 3]
```diff
@@ -83,65 +82,86 @@ class ActionHandler @Inject()(applications: ApplicationOps, applicationForms: Ap
     }
   }
 
-  def RedirectToPreview(id: ApplicationId, sectionNumber: Int): Future[Result] = {
-    Future.successful(Redirect(controllers.routes.ApplicationPreviewController.previewSection(id, sectionNumber)))
-  }
+  def CompleteAndPreview(id: ApplicationId, sectionNumber: Int, fieldValues: JsObject): Future[Result] = {
```
Comment (DougC) [inline src/main/scala/controllers/ActionHandler.scala]: method and function names should start with a lower-case letter

File: src/main/scala/controllers/ActionHandler.scala [hunk 4]
```diff
@@ -83,65 +82,86 @@ class ActionHandler @Inject()(applications: ApplicationOps, applicationForms: Ap
...
+              case errs => redisplaySectionForm(id, sectionNumber, JsonHelpers.flatten("", fieldValues), errs)
+            }
+          }
+        }
+        else redisplaySectionForm(id, sectionNumber, JsonHelpers.flatten("", fieldValues), errs)
+      case CostSection => Future.successful(wip(sectionFormCall(id, sectionNumber).url))
     }
   }
 
-  def redisplaySectionForm(id: ApplicationId, sectionNumber: Int, answers: Map[String, String], errs: FieldErrors = noErrors): Future[Result] = {
```
Comment (DougC) [inline src/main/scala/controllers/ActionHandler.scala]: A lot of this change just seems to be whitespace, or am I missing some details?

File: src/main/scala/controllers/ButtonAction.scala [hunk 5]
```diff
@@ -1,5 +1,7 @@
 package controllers
 
+import play.Logger
```
Comment (BenKDishman) [inline src/main/scala/controllers/ButtonAction.scala]: okay opt-cmd-o.

File: src/main/scala/controllers/ActionHandler.scala (hunk 3 above)
Comment (BenKDishman) [inline src/main/scala/controllers/ActionHandler.scala]: ouch...

File: src/main/scala/controllers/ActionHandler.scala (hunk 4 above)
Comment (BenKDishman) [inline src/main/scala/controllers/ActionHandler.scala]: I think so.  I was wondering why it showed up. Maybe from a new opt-cmd-l ?

Comment (DougC) [inline src/main/scala/controllers/ActionHandler.scala]: Can you just check your IDEA settings - look at Editor->Code Style->Scala and check "Tabs and Indents". Should be don't use tabs and the indents should be 2

**Comment by DougC** [comment]
Okay, cool. Looks like you sorted out the whitespace changes :-)

**Comment by DougC** [comment]
I'm happy to merge once the checks are done

**Pull request merged**